tripal_chado/tests/src/Functional/Services/ChadoTripalPublishTest.php

```diff
@@ -0,0 +1,590 @@
+    $entities = $publish->publish();
+    $this->assertTrue(count($entities) == 0,
+      'The TripalPublish service should return 0 entities when no records are available.');
+
+    //
+    // Test publishing a single record.
+    //
+    $taxrank_db = $idsmanager->loadCollection('TAXRANK', "chado_id_space");
+    $subspecies_term_id = $taxrank_db->getTerm('0000023')->getInternalId();
+
+    $organism_id = $this->addChadoOrganism($chado, [
+      'genus' => 'Oryza',
+      'species' => 'species',
+      'abbreviation' => 'O. sativa',
+      'type_id' => $subspecies_term_id,
+      'infraspecific_name' => 'Japonica',
+      'comment' => 'rice is nice'
+    ]);
+    $entities = $publish->publish();
+    $this->assertTrue(count(array_keys($entities)) == 1,
+      'The TripalPublish service should have published 1 organism.');
+
+    // Test that entries were added for all field items and that fields that
+    // shouldn't be saved in Drupal are NULL.
+    $this->checkFieldItem('organism', 'organism_genus', 1,
+        ['record_id' => $organism_id],
+        ['bundle' => 'organism', 'entity_id' => 1, 'value' => NULL]);
+
+    $this->checkFieldItem('organism', 'organism_species', 1,
+        ['record_id' => $organism_id],
+        ['bundle' => 'organism', 'entity_id' => 1, 'value' => NULL]);
+
+    $this->checkFieldItem('organism', 'organism_abbreviation', 1,
+        ['record_id' => $organism_id],
+        ['bundle' => 'organism', 'entity_id' => 1, 'value' => NULL]);
+
+    $this->checkFieldItem('organism', 'organism_infraspecific_name', 1,
+        ['record_id' => $organism_id],
+        ['bundle' => 'organism', 'entity_id' => 1, 'value' => NULL]);
+
+    $this->checkFieldItem('organism', 'organism_infraspecific_type', 1,
+        ['record_id' => $organism_id],
+        ['bundle' => 'organism', 'entity_id' => 1, 'type_id' => NULL]);
+
+    $this->checkFieldItem('organism', 'organism_comment', 1,
+        ['record_id' => $organism_id],
+        ['bundle' => 'organism', 'entity_id' => 1, 'value' => NULL]);
+
+    // Test that the title via token replacement is working.
+    $this->assertTrue(array_values($entities)[0] == '<em>Oryza species</em> subspecies <em>Japonica</em>',
+        'The title of a Chado organism is incorrect after publishing: ' . array_values($entities)[0] . '!=' . '<em>Oryza species</em> subspecies <em>Japonica</em>');
+
+    //
+    // Test a second entity. Also use a title without all tokens
+    //
+    $organism_id2 = $this->addChadoOrganism($chado, [
+      'genus' => 'Gorilla',
+      'species' => 'gorilla',
+      'abbreviation' => 'G. gorilla',
+      'comment' => 'Gorilla'
+    ]);
+    $entities = $publish->publish();
+    $this->assertTrue(array_values($entities)[0] == '<em>Gorilla gorilla</em> <em></em>',
+        'The title of Chado organism with missing tokens is incorrect after publishing: "' . array_values($entities)[0] . '" != "<em>Gorilla gorilla</em> <em></em>"');
```

Comment:
It's better to use assert equals for this comparison because it provides more information if there is a failure. This way the error message will include the comparison just like you tried to add it into the message. So in this exact case it's equivalent but the formatting is better and it's easier to make sure the values shown don't get out sync in the future.
```suggestion
    $this->assertEquals('<em>Gorilla gorilla</em> <em></em>', array_values($entities)[0],
        'The title of Chado organism with missing tokens is incorrect after publishing');
```